File: Core_PetBoarding_Backend/PetBoarding_Api/Dto/Pets/PetDto.cs
using PetBoarding_Domain.Pets;

namespace PetBoarding_Api.Dto.Pets;

public record PetDto
{
    public Guid Id { get; init; }
    public string Name { get; init; } = string.Empty;
    public PetType Type { get; init; }
    public string Breed { get; init; } = string.Empty;
    public int Age { get; init; }
    public decimal? Weight { get; init; }
    public string Color { get; init; } = string.Empty;
    public PetGender Gender { get; init; }
    public bool IsNeutered { get; init; }
    public string? MicrochipNumber { get; init; }
    public string? MedicalNotes { get; init; }
    public string? SpecialNeeds { get; init; }
    public string? PhotoUrl { get; init; }
    public Guid OwnerId { get; init; }
    public string? OwnerName { get; init; }
    public EmergencyContactDto? EmergencyContact { get; init; }
    public DateTime CreatedAt { get; init; }
    public DateTime UpdatedAt { get; init; }
}

File: ng_PetBoarding_app/src/app/shared/contracts/pets/pet.dto.ts
export interface PetDto {
  id: string;
  name: string;
  type: number; // Backend utilise des integers pour PetType
  breed: string;
  age: number;
  weight?: number;
  color: string;
  gender: number; // Backend utilise des integers pour PetGender
  isNeutered: boolean;
  microchipNumber?: string;
  medicalNotes?: string;
  specialNeeds?: string;
  photoUrl?: string;
  ownerId: string;
  ownerName?: string;
  emergencyContact?: EmergencyContactDto;
  createdAt: string;
  updatedAt: string;
}

export interface EmergencyContactDto {
  name: string;
  phone: string;
  relationship: string;
}

export interface CreatePetRequest {
  name: string;
  type: number;
  breed: string;
  age: number;
  weight?: number;
  color: string;
  gender: number;
  isNeutered: boolean;
  microchipNumber?: string;
  medicalNotes?: string;
  specialNeeds?: string;
  photoUrl?: string;
  emergencyContactName?: string;
  emergencyContactPhone?: string;
  emergencyContactRelationship?: string;
}

export interface UpdatePetRequest {
  name: string;
  type: number;
  breed: string;
  age: number;
  weight?: number;
  color: string;
  gender: number;
  isNeutered: boolean;
  microchipNumber?: string;
  medicalNotes?: string;
  specialNeeds?: string;
  photoUrl?: string;
  emergencyContactName?: string;
  emergencyContactPhone?: string;
  emergencyContactRelationship?: string;
}

export interface GetAllPetsResponse {
  pets: PetDto[];
  count: number;
  success: boolean;
  message: string;
}

export interface GetPetResponse {
  pet: PetDto;
  success: boolean;
  message: string;
}

export interface CreatePetResponse {
  pet: PetDto;
  success: boolean;
  message: string;
  location?: string;
}

export interface UpdatePetResponse {
  pet: PetDto;
  success: boolean;
  message: string;
}

export interface DeletePetResponse {
  success: boolean;
  message: string;
  petId?: string;
}